runtime/reconcilers/metrics_view.go

```diff
@@ -108,6 +120,16 @@ func (r *MetricsViewReconciler) Reconcile(ctx context.Context, n *runtimev1.Reso
 		return runtime.ReconcileResult{Err: fmt.Errorf("failed to create metrics view executor: %w", err)}
 	}
 	defer e.Close()
+
+	reconciled, err := e.ReconcileWithParentMetricsView(ctx)
+	if err != nil {
+		return runtime.ReconcileResult{Err: fmt.Errorf("failed to normalize parent metrics view: %w", err)}
+	}
+	if reconciled != nil {
+		// If the parent metrics view was normalized, we update the spec to match.
+		mv.Spec = reconciled
+	}
+
 	validateResult, validateErr := e.ValidateAndNormalizeMetricsView(ctx)
```

Comment:
Something feels a little weird here:
1. The executor was created with `mv.Spec`, not the rewritten spec, so you'd expect to have to create a new executor for the rewritten spec before calling `ValidateAndNormalizeMetricsView`.
2. Would it make sense to put parent metrics view handling into `ValidateAndNormalizeMetricsView`? Since it already says it "normalizes" the metrics view, and handling parents is really just a kind of normalization? Or are there advantages to decoupling it?

Reply:
Moved resolution of parent metrics view inside `ValidateAndNormalizeMetricsView`